src/handlers/ens.ts

```diff
@@ -771,11 +802,26 @@ export const estimateENSSetRecordsGasLimit = async ({
     const shouldUseMulticall = shouldUseMulticallTransaction(
       ensRegistrationRecords
     );
-    gasLimit = await (shouldUseMulticall
-      ? estimateENSMulticallGasLimit
-      : estimateENSSetTextGasLimit)({
-      ...{ name, ownerAddress, records: ensRegistrationRecords },
-    });
+    if (shouldUseMulticall) {
+      gasLimit = await estimateENSMulticallGasLimit({
+        name,
+        ownerAddress,
+        records: ensRegistrationRecords,
+      });
+    } else {
+      if (ensRegistrationRecords?.coinAddress?.length) {
```

Comment:
nice setAddress addition 👌 

setAddress is meant to just one coin, if there are 2 coinAddress we'll be trying to send setAddr txs. So if there's more we should use multicall, maybe you could add this condition to `shouldUseMulticallTransaction` ? so if there's only one coinAddress or only one text record we should use setAddr or setText respectively 

WDYT?

Reply:
since we have 3 types of txs now maybe we should replace `shouldUseMulticallTransaction` to a method that returns the type of tx that we're gonna use? to make it cleaner like

```
const getRecordsTypeTransaction = () => {
  if (more than one text record and/or coinAddress) return MULTICALL
  if (only one text record) return SET_TEXT
  if (only one coinAddress) return SET_ADDR
}
```
and then estimate the gas corresponding on the records tx type

Reply:
yeah thats a good call